security-analysis/security-analysis-api/src/test/java/com/powsybl/security/SecurityAnalysisToolTest.java

```diff
@@ -45,10 +68,64 @@ public void assertCommand() {
         assertOption(tool.getCommand().getOptions(), "task-count", false, true);
         assertOption(tool.getCommand().getOptions(), "task", false, true);
         assertOption(tool.getCommand().getOptions(), "external", false, false);
+        assertOption(tool.getCommand().getOptions(), "log-file", false, true);
     }
 
     @Test
     public void test() {
         assertCommand();
     }
+
+    @Test
+    public void testRunWithLog() throws Exception {
+        ByteArrayOutputStream bout = new ByteArrayOutputStream();
+        ByteArrayOutputStream berr = new ByteArrayOutputStream();
+        PrintStream out = new PrintStream(bout);
+        PrintStream err = new PrintStream(berr);
+        CommandLine cl = mock(CommandLine.class);
+        ToolRunningContext context = mock(ToolRunningContext.class);
+        when(context.getFileSystem()).thenReturn(fileSystem);
+        when(context.getOutputStream()).thenReturn(out);
+        when(context.getErrorStream()).thenReturn(err);
+        when(cl.getOptionValue("case-file")).thenReturn("network.xml");
+        when(cl.hasOption("limit-types")).thenReturn(false);
+        when(cl.getOptionProperties(any())).thenReturn(new Properties());
+        // tigger runWithLog()
+        when(cl.hasOption("log-file")).thenReturn(true);
+        when(cl.getOptionValue("log-file")).thenReturn("out.zip");
+        ComputationManager cm = mock(ComputationManager.class);
+        when(context.getShortTimeExecutionComputationManager()).thenReturn(cm);
+        when(context.getLongTimeExecutionComputationManager()).thenReturn(cm);
+        mockStatic(Importers.class);
+        ImportConfig config = ImportConfig.load();
+        Network network = mock(Network.class);
+        when(network.getVariantManager()).thenReturn(mock(VariantManager.class));
+        BDDMockito.given(Importers.loadNetwork(any(Path.class), any(ComputationManager.class), any(ImportConfig.class), any(Properties.class))).willReturn(network);
```

Comment:
Is it thread safe? In case of parallel unit run.

Reply:
Why do you need BDDMockito? In PowerMockito wiki, there is an example https://github.com/powermock/powermock/wiki/MockStatic

Reply:
I don't know which version is used in his wiki. In the version I used, there is no such method and I can't change the powermock's version either because of the compatibility with the current mockito.